polyphemus/spec/cascade_mvir_restricted_spec.rb

```diff
@@ -0,0 +1,88 @@
+describe Polyphemus::CascadeMvirPatientWaiverToRestricted do
+  let(:command) { Polyphemus.instance.commands[:cascade_mvir_patient_waiver_to_restricted] }
+
+  let(:all_patients) {
+    {
+       'Dave': { name: 'Dave', restricted: true, consent: 'Full Study' },
+       'Sally': { name: 'Sally', restricted: false, consent: 'Full Study' },
+       'Dan': { name: 'Dan', restricted: false, consent: 'Initial Waiver' },
+       'Fred': { name: 'Fred', restricted: true, consent: 'Initial Waiver' },
+       'Mike': { name: 'Mike', restricted: false, consent: 'No Samples/Data' },
+    }
+  }
+
+  let(:all_pools) { ['pool-a', 'pool-b'] }
+  let(:restricted_pools) { ['pool-a'] }
+
+  before(:each) do
+    @all_updates = []
+
+    WebMock.disable_net_connect!
+    stub_request(:post, 'https://magma.test/retrieve')
+        .with(body: hash_including({ project_name: 'mvir1', model_name: 'patient',
+                                     attribute_names: 'all', record_names: 'all' }))
+        .to_return({ body: {
+            'models': { 'patient': { 'documents': all_patients } }
+        }.to_json })
+
+    stub_request(:post, 'https://magma.test/update')
+        .to_return do |request|
+
+      body = StringIO.new(request.body)
+      content_length = body.read.length
+      body.rewind
+
+      tempfile = Rack::Multipart::Parser::TEMPFILE_FACTORY
```

Comment:
I reaaallly wish that multipart parsing were simpler for testing.  Webmock straight up errors and warns it does not support it.  So I had to hack this bit in.